packages/perseus/src/widgets/interactive-graphs/graphs/utils.ts

```diff
@@ -18,35 +18,31 @@ export const getIntersectionOfRayWithBox = (
     box: [x: Interval, y: Interval],
 ): [number, number] => {
     const [[xMin, xMax], [yMin, yMax]] = box;
-    const [aX, aY] = throughPoint;
-    const [bX, bY] = initialPoint;
+    const [aX, aY] = initialPoint;
+    const [bX, bY] = throughPoint;
 
     const yDiff = bY - aY;
     const xDiff = bX - aX;
     const slope = yDiff / xDiff;
+    const inverseSlope = 1 / slope;
 
-    const yAtXMin = slope * (xMin - aX) + aY;
-    const yAtXMax = slope * (xMax - aX) + aY;
-    const xAtYMin = (yMin - aY) / slope + aX;
-    const xAtYMax = (yMax - aY) / slope + aX;
+    const xExtreme = xDiff < 0 ? xMin : xMax;
+    const yExtreme = yDiff < 0 ? yMin : yMax;
+
+    const yAtXExtreme = aY + (xExtreme - aX) * slope;
+    const xAtYExtreme = aX + (yExtreme - aY) * inverseSlope;
 
-    // clock analogy to describe quadrants
     switch (true) {
-        // 12 o'clock to 2:59
-        case yDiff > 0 && xDiff >= 0:
-            return xAtYMax > xMax ? [xMax, yAtXMax] : [xAtYMax, yMax];
-        // 3 o'clock to 5:59
-        case yDiff <= 0 && xDiff > 0:
-            // xAtYMin evaluates to -Infinity here, so we use absolute value
-            return Math.abs(xAtYMin) > xMax ? [xMax, yAtXMax] : [xAtYMin, yMin];
-        // 9 o'clock to 11:59
-        case yDiff >= 0 && xDiff < 0:
-            return xAtYMax < xMin ? [xMin, yAtXMin] : [xAtYMax, yMax];
-        // 6 o'clock to 8:59
-        case yDiff < 0 && xDiff <= 0:
-            return xAtYMin < xMin ? [xMin, yAtXMin] : [xAtYMin, yMin];
+        // does the ray intersect the left or right edge?
```

Comment:
I'm not sure if it's obvious to others, but this is: 

```suggestion
        // does the ray intersect the left or right edge within the y graph range?
```

right?